addons/lunch/static/src/js/lunch_kanban_widget.js

```diff
@@ -0,0 +1,169 @@
+odoo.define('lunch.LunchKanbanWidget', function (require) {
+"use strict";
+
+var core = require('web.core');
+var dialogs = require('web.view_dialogs');
+var field_registry = require('web.field_registry');
+var session = require('web.session');
+var Widget = require('web.Widget');
+
+var _t = core._t;
+var qweb = core.qweb;
+var FieldMany2One = field_registry.get('many2one');
+
+var LunchKanbanWidget = Widget.extend({
+    template: 'LunchKanbanWidget',
+    custom_events: {
+        'field_changed': '_onFieldChanged',
+    },
+    events: {
+        'click .o_add_money': '_onAddMoney',
+        'click .o_add_product': '_onAddProduct',
+        'click .o_lunch_widget_order_button': '_onOrderNow',
+        'click .o_remove_product': '_onRemoveProduct',
+        'click .o_lunch_widget_save': '_onSaveOrder',
+        'click .o_lunch_widget_unlink': '_onUnlinkOrder',
+        'click .o_lunch_open_wizard': '_onLunchOpenWizard',
+    },
+
+    init: function (parent, params) {
+        var self = this;
+        this._super.apply(this, arguments);
+
+        this.is_manager = params.is_manager || false;
+        this.userimage = params.userimage || '';
+        this.username = params.username || '';
+
+        this.lunchUserField = null;
+
+        this.group_portal_id = undefined;
+
+        self._rpc({
```

Comment:
this does not work, the then will execute after the end of the function so self.group_portal_id will not be set immediately

Reply:
Also, RPC's in the init function is really not a good idea, you should put it in the willStart